firefox-ios/Client/Coordinators/Router/RouteBuilder.swift

```diff
@@ -63,7 +57,7 @@ final class RouteBuilder {
             case .openUrl:
                 // If we have a URL query, then make sure to check its a webpage
                 if urlQuery == nil || urlQuery?.isWebPage() ?? false {
-                    return .search(url: urlQuery, isPrivate: isPrivate)
+                    return .search(url: urlQuery, isPrivate: false)
```

Comment:
for my better understanding, can you help clarify why this change and why only these places are set to false?

Reply:
Previously they would open the app into private mode if the previous session was a private session. But since we no longer store the private tab data the expected session type on launch is normal browsing (unless the user explicitly uses the widget or shortcut to open a private tab).

Reply:
> since we no longer store the private tab data the expected session type on launch is normal browsing

Some of these codepaths are also called when the app is simply in the background, however (as opposed to a cold launch). And in those cases private tabs are still active. The widget routing can run with private tabs open and at least in a quick test I could see this .`openURL` case handled when a private tab was still open/active. At first glance I don't think we should be hardcoding these to `false` 🤔

Reply:
Yeah I misunderstood how this works, the tests caught the issue. I've just pushed an update with this fixed. It no longer passes in the previous private session but does still look through the url for the private parameter.